pkg/pillar/containerd/containerd.go

```diff
@@ -706,35 +710,43 @@ func createMountPointExecEnvFiles(containerPath string, mountpoints map[string]s
 // it R/O into the container. On top of that we expect the usual suspects of /run,
 // /persist and /config to be taken care of by the OCI config that lk produced.
 func LKTaskLaunch(name, linuxkit string, domSettings *types.DomainConfig, args []string) (int, error) {
-	config := "/containers/services" + linuxkit + "/config.json"
-	rootfs := "/containers/services" + linuxkit + "/lower"
+	config := "/containers/services/" + linuxkit + "/config.json"
+	rootfs := "/containers/services/" + linuxkit + "/rootfs"
```

Comment:
If you need a R/W filesystem you need to prepare a volume for it. A better question is: why DO you need a R/W root filesystem? All the R/W bits are given to you via /run and /persist anyway -- so what's missing?

Reply:
I had to make rootfs as R/W because of  /dev/, /sys, /proc systems, since qemu/kernel is creating /dev/pts/ and  /proc/ files during the VM launch.

Reply:
If those are the only ones -- we need to dig deeper into this as well. Those are supposed to be R/W mounted from the outside when you ask runc to run the container. Lets start with seeing what gets mounted on those endpoint when you call LKTaskLaunch() and we can proceed from there.

Reply:
Hi @rvs, you are right, I unnecessarily marked this as R/W, whereas Readonly rootfs works just fine, as I verified it today.

Reply:
Hi @rvs, I already reverted the Root.Readonly = false part.

Reply:
The only issue with read-only root partition is that, we can't use "screen" with default settings, 
since it tries to create /tmp/.screens directory, and it fails. But we can always override that by 
explicitly mentioning "SCREENDIR" env variable. So we are all good with the Read-only setting.